kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternDetailsController.java

```diff
@@ -264,6 +265,10 @@ public void onChanged(Change<? extends DescrName> change) {
         patternFieldsPanelEventSubscriber = evt -> {
             PatternField patternField = evt.getPatternField();
             int fieldPosition = evt.getCurrentFieldOrder()-1;
+            if(evt.getEventType() == EDIT_FIELD){
+                // 1st remove it from list before adding the new entry
+                patternFieldList.remove(patternViewModel.getPropertyValue(SELECTED_PATTERN_FIELD));
```

Comment:
This statement is not needed. The event's patternField is all you need. This subscriber will update the FIELD_COLLECTION (observable list). The FIELD_COLLECTION's listener will do the removing of the UI's VBox item. Remember there are two collections, they are in sync. If you remove and item (by index) it's the same remove. 
Assuming you are doing a remove and an insert(add).

Reply:
Hi Carl, I think that is exactly what the above code is doing. Removing it from the FIELDS_COLLECTION the listener of the FIELDS_COLLECTION is updating the UI.  There is no other place in this controller class code where we are removing the selected pattern. 
    The events patternField is the new value/instance that is add to the FILEDS_COLLECTION list and to the UI via its listener. It is not related or same as the to the SELECTED_PATTER_FIELD value that we are removing. They are 2 separate instances. 
  Yes, I am doing remove and then insert/add only if the event type is EDIT_FIELD.
   Else I am doing only insert/add if the event type is ADD_FIELD.